### PR TITLE
Update migration guide

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/media-migration.md [hunk 1]
```diff
@@ -1,151 +1,278 @@
-# Migrate from Media Analytics SDK to AEP SDK on Roku platform
+# Migrate from Media SDK to AEPRoku SDK
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:1]: Should we change `AEPRoku SDK` to `AEP Roku SDK`?

File: Documentation/media-migration.md [hunk 2]
```diff
@@ -1,151 +1,278 @@
...
+
+## Prerequisistes
+1. [Experience Data Model (XDM)](https://experienceleague.adobe.com/docs/experience-platform/xdm/home.html?lang=en)
+2. [Datastreams](https://developer.adobe.com/client-sdks/home/getting-started/configure-datastreams/)
+3. Setup schemas, datastream, dataset, Customer Journey Analytics (CJA) dashboard etc. using this [guide](https://experienceleague.adobe.com/docs/media-analytics/using/implementation/edge-recommended/media-edge-sdk/implementation-edge.html) for Media tracking using AEPEdge SDK.
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:6]: `using this guide for Media tracking using AEPEdge SDK.` 
Should we reword it?

File: Documentation/media-migration.md [hunk 3]
```diff
@@ -1,151 +1,278 @@
...
+
+## Table of contents
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:8]: nit: should we move the `Prerequisistes` section after the TOC?

File: Documentation/media-migration.md [hunk 4]
```diff
@@ -1,151 +1,278 @@
...
+| `adb_media_init_chapterinfo(title, position, length, startTime)` | NA |
+| `adb_media_init_qosinfo(bitrate, startupTime, fps, droppedFrames)` | NA |
+
+## Initliaze SDK instance
+
+**Media SDK**
+
+```brightscript
+' Create adbmobileTask node
+m.adbmobileTask = createObject("roSGNode","adbmobileTask") <br> Get AdobeMobile SG connector instance
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:92]: remove `<br> Get AdobeMobile SG connector instance`

File: Documentation/media-migration.md [hunk 5]
```diff
@@ -1,151 +1,278 @@
...
+```
 
-If you are currently using the Media Analytics SDK for your Roku projects, this guide offers a detailed outline on how to migrate to AEP Roku SDK using a step-by-step approach.
+**AEPRoku SDK**
 
-In AEP Roku SDK, there are two Media trakcing APIs:
+```brightscript
+' Create SDK instance
+m.aepSdk = AdobeAEPSDKInit()
+```
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:102]: Can we add a link to the `getting started` doc?

File: Documentation/media-migration.md [hunk 6]
```diff
@@ -1,151 +1,278 @@
...
-        }
-      }
+sessionDetails = {
+    "streamType": "video",
+    "friendlyName": "test_media_name",
+    "name": "test_media_id",
+    "length": 10,
+    "contentType": "vod",
+
+    ''' (Optional) Attach Standard metadata if any
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:142]: `''' (Optional) Attach Standard metadata if any` -> `' (Optional) Attach Standard metadata if any`
https://developer.roku.com/docs/references/brightscript/language/expressions-variables-types.md#in-brightscript
Can we use only one apostrophe (') here?

File: Documentation/media-migration.md [hunk 7]
```diff
@@ -1,151 +1,278 @@
...
+| Sections |
+| -- |
+| [API comparison between SDKs](#api-comparison) |
+| [Initliaze SDK instance](#initliaze-sdk-instance) |
+| [Start Media session](#start-media-session) |
+| [Track Media events](#track-media-events) |
+
+## API comparison
+
+### Core Plaback APIs:
```
Comment (yangyansong-adbe) [inline Documentation/media-migration.md:18]: Comparing APIs for different tracking scenarios is made extremely clear.

File: Documentation/media-migration.md (hunk 5 above)
Comment (addb) [reply Documentation/media-migration.md:102]: Should we add it in prereq?

Comment (addb) [reply Documentation/media-migration.md:105]: Added in prereq. Let me know if your thoughts.

Comment (yangyansong-adbe) [reply Documentation/media-migration.md:106]: Yup, it works.

File: Documentation/media-migration.md [hunk 8]
```diff
@@ -1,151 +1,282 @@
...
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, adBreakInfo, contextData)` | `sendMediaEvent(adbreakStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, invalid, invalid)` | `sendMediaEvent(adbreakCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_START, adInfo, contextData)` | `sendMediaEvent(adStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_COMPLETE, invalid, invalid)` | `sendMediaEvent(adCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_SKIP, invalid, invalid)` | `sendMediaEvent(adSkipXDM)` |
+
+### Buffer and Seek APIs
+| Media SDK | AEP SDK|
+| -- | -- |
+| `mediaTrackEvent(ADBMobile().MEDIA_BUFFER_START, bufferInfo, bufferContextData)` | `sendMediaEvent(bufferCompleteXDM)` |
```
Comment (praveek) [inline Documentation/media-migration.md:46]: The AEP section is incorrect.

File: Documentation/media-migration.md [hunk 9]
```diff
@@ -1,151 +1,282 @@
...
+| `mediaTrackPlay()` | `sendMediaEvent(playXDM)` |
+| `mediaTrackPause()` | `sendMediaEvent(pauseStartXDM)` |
+| `mediaTrackComplete()` | `sendMediaEvent(sessionCompleteXDM)` |
+| `mediaTrackSessionEnd()` | `sendMediaEvent(sessionEndXDM)` |
+
+### Ad Tracking APIs:
+| Media SDK | AEP SDK|
+| -- | -- |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, adBreakInfo, contextData)` | `sendMediaEvent(adbreakStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, invalid, invalid)` | `sendMediaEvent(adbreakCompleteXDM)` |
```
Comment (praveek) [inline Documentation/media-migration.md:38]: MEDIA_AD_BREAK_COMPLETE

File: Documentation/media-migration.md [hunk 10]
```diff
@@ -1,151 +1,282 @@
...
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_START, adInfo, contextData)` | `sendMediaEvent(adStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_COMPLETE, invalid, invalid)` | `sendMediaEvent(adCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_SKIP, invalid, invalid)` | `sendMediaEvent(adSkipXDM)` |
+
+### Buffer and Seek APIs
+| Media SDK | AEP SDK|
+| -- | -- |
+| `mediaTrackEvent(ADBMobile().MEDIA_BUFFER_START, bufferInfo, bufferContextData)` | `sendMediaEvent(bufferCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_BUFFER_COMPLETE, invalid, invalid)` | `sendMediaEvent(bufferCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_SEEK_START, seekInfo, seekContextData)` | `sendMediaEvent(pauseStartXDM)` |
```
Comment (praveek) [inline Documentation/media-migration.md:48]: For Media SDK, we don't have to provide seekInfo or seekContextData

File: Documentation/media-migration.md [hunk 11]
```diff
@@ -1,151 +1,282 @@
...
+| -- | -- |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, adBreakInfo, contextData)` | `sendMediaEvent(adbreakStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_BREAK_START, invalid, invalid)` | `sendMediaEvent(adbreakCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_START, adInfo, contextData)` | `sendMediaEvent(adStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_COMPLETE, invalid, invalid)` | `sendMediaEvent(adCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_SKIP, invalid, invalid)` | `sendMediaEvent(adSkipXDM)` |
+
+### Buffer and Seek APIs
+| Media SDK | AEP SDK|
+| -- | -- |
```
Comment (praveek) [inline Documentation/media-migration.md:45]: For Media SDK, we don't have to provide bufferInfo or bufferContextData

Comment (addb) [reply Documentation/media-migration.md:45]: Lot of is copy paste from our Media Docs and I dont know what happened but most of the APIs are wrong in the  experience league doc for Media. I corrected some but was going to refine it. Thanks for highlighting.

File: Documentation/media-migration.md [hunk 12]
```diff
@@ -1,220 +1,828 @@
...
+1. [Experience Data Model (XDM)](https://experienceleague.adobe.com/docs/experience-platform/xdm/home.html?lang=en)
+2. [Datastreams](https://developer.adobe.com/client-sdks/home/getting-started/configure-datastreams/)
+3. [Getting Started with implementing Media tracking using AEP Roku SDK](./getting-started.md)
+4. [Setup schemas, datastream, dataset, Customer Journey Analytics (CJA) dashboard etc](https://experienceleague.adobe.com/docs/media-analytics/using/implementation/edge-recommended/media-edge-sdk/implementation-edge.html)
+
+
+## API comparison
+
+> [!NOTE]
+> AEP SDK has only two media APIs `createMediaSession()` and `sendMediaEvent()`.
```
Comment (praveek) [inline Documentation/media-migration.md:22]: APIs for tracking media

File: Documentation/media-migration.md [hunk 13]
```diff
@@ -1,220 +1,828 @@
...
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_COMPLETE, invalid, invalid)` | `sendMediaEvent(adCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_AD_SKIP, invalid, invalid)` | `sendMediaEvent(adSkipXDM)` |
+
+### Buffer and Seek APIs
+| Media SDK | AEP SDK|
+| -- | -- |
+| `mediaTrackEvent(ADBMobile().MEDIA_BUFFER_START, invalid, invalid)` | `sendMediaEvent(bufferStartXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_BUFFER_COMPLETE, invalid, invalid)` | `sendMediaEvent(bufferCompleteXDM)` |
+| `mediaTrackEvent(ADBMobile().MEDIA_SEEK_START, invalid, invalid)` | `sendMediaEvent(pauseStartXDM)` |
+| `ADBMobile().mediaTrackEvent(ADBMobile().MEDIA_SEEK_COMPLETE, invalid, invalid)` | `sendMediaEvent(pauseStartXDM)` |
```
Comment (praveek) [inline Documentation/media-migration.md:49]: `sendMediaEvent(playXDM)`

File: Documentation/media-migration.md [hunk 14]
```diff
@@ -1,220 +1,828 @@
...
+
+> [!NOTE]
```
Comment (praveek) [inline Documentation/media-migration.md:51]: Can you move this note below the heading? Also reword 
`Media backend will detect seek based on change in playhead and timestamp values.`

File: Documentation/media-migration.md [hunk 15]
```diff
@@ -1,220 +1,828 @@
...
+    "mediaCollection": {
+      "playhead": <CURRENT_PLAYHEAD_INTEGER_VALUE>
+    }
+  }
+}
+
+m.aepSdk.sendMediaEvent(adSkipXDM)
+```
+
+#### AdbreakComplete
```
Comment (praveek) [inline Documentation/media-migration.md:429]: nit: Move AdbreakComplete next to AdbreakStart

File: Documentation/media-migration.md [hunk 16]
```diff
@@ -1,220 +1,828 @@
...
+> [!NOTE]
+> To learn more refer to the [errorDetails](https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/errordetails.schema.md) XDM fieldgroup.
+
+## Public Constants
+
+### StreamType
+
+| Media SDK | AEP SDK |
+| -- | -- |
+| MEDIA_STREAM_TYPE_VOD | NA |
```
Comment (praveek) [inline Documentation/media-migration.md:773]: Can you provide the string value here?

File: Documentation/media-migration.md [hunk 17]
```diff
@@ -1,220 +1,828 @@
...
+| MEDIA_STREAM_TYPE_LIVE | NA |
+| MEDIA_STREAM_TYPE_LINEAR | NA |
+| MEDIA_STREAM_TYPE_AOD | NA |
+| MEDIA_STREAM_TYPE_AUDIOBOOK | NA |
+| MEDIA_STREAM_TYPE_PODCAST | NA |
+
+### MediaType
+
+| Media SDK | AEP SDK |
+| -- | -- |
```
Comment (praveek) [inline Documentation/media-migration.md:783]: Can you provide the string value here?

File: Documentation/media-migration.md [hunk 18]
```diff
@@ -1,220 +1,828 @@
...
+| MEDIA_STREAM_TYPE_AUDIO | NA |
+| MEDIA_STREAM_TYPE_VIDEO | NA |
+
+### Standard Video Metadata
+
+| Media SDK | AEP SDK |
+| -- | -- |
+| MEDIA_VideoMetadataKeySHOW                    | Refer to [sessionDetails](https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/sessiondetails.schema.md) XDM fieldgroup |
```
Comment (praveek) [inline Documentation/media-migration.md:791]: Can you provide the link to the correct field?
https://github.com/adobe/xdm/blob/master/docs/reference/datatypes/sessiondetails.schema.md#xdmshow

File: Documentation/media-migration.md (hunk 17 above)
Comment (addb) [reply Documentation/media-migration.md:783]: Yes

**Pull request merged**